### DISCUSSION TITLE
Deny everything BUT a topic ?

### DISCUSSION THREAD

**Discussion by ZacharieGignac** [emqx/emqx#8085] [General]
Trying to wrap my head around this problem I have.
I have to give access to certain usernames, in thier own "topic", but deny them access to everything else.

For example, username "Alice" needs access to "systems/alice/#" and "Bob" needs access to "systems/bob/#". I need them to be completely "isolated" in thier own path.

Even if I set the pubsub for Alice for systems/alice/#, this user can still publish everywhere it wants.

How can I accomplish what I need ? I need to be able to DENY everything that does not match my allows.

Thanks for the help!

**Comment by ZacharieGignac** [comment]
Missing that information -> using mnesia.

**Reply by lafirest** [reply]
If I understand correctly, you need to add one more `deny` rule, like this:
<img width="1373" alt="image" src="https://user-images.githubusercontent.com/22971499/173272939-1511c388-4a6b-4291-8f58-f58bb5484bde.png">
